Bottom: 80af375fecf24d84c6f9320e1c981088d1a140a2
Top:    79c80554ca72185ed4bcfa02887fb24a98f5755b
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-06-08 11:29:53 -0500

Refresh of all-docker_image

---
diff --git a/ansible-playbook/group_vars/docker-hosts/docker.yml b/ansible-playbook/group_vars/docker-hosts/docker.yml
index f9faae1c..0d394562 100644
--- a/ansible-playbook/group_vars/docker-hosts/docker.yml
+++ b/ansible-playbook/group_vars/docker-hosts/docker.yml
@@ -21,3 +21,6 @@ docker_restart_policy: 'always'
 
 # Network names
 docker_network_frontend: 'frontend'
+
+# docker force build images
+docker_image_force_build: true
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_buildsystem.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_buildsystem.yml
index 34167b2f..3a0496ca 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_buildsystem.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_buildsystem.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_http_data_buildsystem_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_http_data_buildsystem.dir_name }}"
     state: "{{ image_build_state_data }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station build system data
   docker_container: 
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_database.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_database.yml
index 3f6150c4..ef3f24ba 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_database.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_database.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_database_data_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_database_data.dir_name }}"
     state: "{{ image_build_state_data }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station database data
   docker_container: 
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ipkg.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ipkg.yml
index 1c867665..e167a48c 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ipkg.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ipkg.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_http_data_ipkg_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_http_data_ipkg.dir_name }}"
     state: "{{ image_build_state_data }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station ipkg data
   docker_container: 
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_iso.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_iso.yml
index 27421a72..4e672f6e 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_iso.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_iso.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_http_data_ncdiso_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_http_data_ncdiso.dir_name }}"
     state: "{{ image_build_state_data }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station ncd iso data
   docker_container: 
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ncdrelease.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ncdrelease.yml
index 0c950bab..f0d2c7e4 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ncdrelease.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_ncdrelease.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_http_data_ncdrelease_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_http_data_ncdrelease.dir_name }}"
     state: "{{ image_build_state_data }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station ncd release data
   docker_container: 
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_testclient.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_testclient.yml
index 47db0a2f..e2157a00 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_testclient.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_testclient.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_http_data_testclient_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_http_data_testclient.dir_name }}"
     state: "{{ image_build_state_data }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station  test client data
   docker_container: 
diff --git a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_tftp.yml b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_tftp.yml
index 8b296f1b..85b16f0d 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/datacontainer_tftp.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/datacontainer_tftp.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_tftp_data_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_tftp_data.dir_name }}"
     state: "{{ image_build_state_data }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station tftp data
   docker_container: 
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_database.yml b/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
index e88455d0..a32f6a8b 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_database.yml
@@ -19,6 +19,7 @@
     name: "{{ name_prefix }}teststation_database_server_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_database_server.dir_name }}"
     state: "{{ image_build_state_nondata }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station database server
   docker_container:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_http.yml b/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
index dea13311..f3af2a3b 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_http.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_http_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_http_server.dir_name }}"
     state: "{{ image_build_state_nondata }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station http server
   docker_container:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml b/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
index 977ac9a1..f49b020f 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_ssh.yml
@@ -13,6 +13,7 @@
     name: "{{ name_prefix }}teststation_ssh_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_ssh_server.dir_name }}"
     state: "{{ image_build_state_nondata }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station ssh server
   docker_container:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml b/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
index 12c95bb1..a1bc6331 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_ssh_prod.yml
@@ -25,6 +25,7 @@
     name: "{{ name_prefix }}teststation_ssh_server_manualtest_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_ssh_server_manualtest.dir_name }}"
     state: "{{ image_build_state_nondata }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station production ssh server temp.
   docker_container:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml b/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
index 714027bf..64ca521e 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/server_tftp.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_tftp_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_tftp_server.dir_name }}"
     state: "{{ image_build_state_nondata }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station tftp server
   docker_container:
diff --git a/roles/docker-buildsystem/tasks/docker_containers/service_backup.yml b/roles/docker-buildsystem/tasks/docker_containers/service_backup.yml
index c0aebb83..6b459ccb 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/service_backup.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/service_backup.yml
@@ -18,5 +18,4 @@
     name: "{{ name_prefix }}teststation_supportsite_backup_restore_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.Backup_Restore.dir_name }}"
     state: "{{ image_build_state_nondata }}"
-
-
+    force: "{{ docker_image_force_build }}"
diff --git a/roles/docker-buildsystem/tasks/docker_containers/service_cron.yml b/roles/docker-buildsystem/tasks/docker_containers/service_cron.yml
index f52909d5..8b0712fc 100644
--- a/roles/docker-buildsystem/tasks/docker_containers/service_cron.yml
+++ b/roles/docker-buildsystem/tasks/docker_containers/service_cron.yml
@@ -7,6 +7,7 @@
     name: "{{ name_prefix }}teststation_cron_image"
     path: "{{ docker_projects_dir }}/{{ build_system_docker_repos.TestStation_cron_service.dir_name }}"
     state: "{{ image_build_state_nondata }}"
+    force: "{{ docker_image_force_build }}"
 
 - name: Start docker container - Test Station cron service
   docker_container:
diff --git a/roles/docker-exim4/tasks/main.yml b/roles/docker-exim4/tasks/main.yml
index 5accd8e7..fa74e754 100644
--- a/roles/docker-exim4/tasks/main.yml
+++ b/roles/docker-exim4/tasks/main.yml
@@ -87,6 +87,7 @@
     name: '{{ exim4_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-exim4'
+    force: "{{ docker_image_force_build }}"
 
 # *****************************************************************************
 # Create the data volumes
diff --git a/roles/docker-gitlab/tasks/main.yml b/roles/docker-gitlab/tasks/main.yml
index 8559fa4c..9eda5465 100644
--- a/roles/docker-gitlab/tasks/main.yml
+++ b/roles/docker-gitlab/tasks/main.yml
@@ -85,6 +85,7 @@
     name: '{{ gitlab_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-gitlab'
+    force: "{{ docker_image_force_build }}"
 
 # Data volume image is now the same as the container image
 #- name: build image
@@ -92,6 +93,7 @@
 #    image_name: '{{ gitlab_dv_image_name }}'
 #    image_tag: '{{ docker_image_tag }}'
 #    dockerfile_dir: '{{ docker_projects_dir }}/docker-gitlab'
+#    force: "{{ docker_image_force_build }}"
 
 # *****************************************************************************
 # Create the data volumes
diff --git a/roles/docker-mantisbt/tasks/main.yml b/roles/docker-mantisbt/tasks/main.yml
index 1a0f82a2..62396985 100644
--- a/roles/docker-mantisbt/tasks/main.yml
+++ b/roles/docker-mantisbt/tasks/main.yml
@@ -84,6 +84,7 @@
     name: '{{ mantisbt_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-mantisbt'
+    force: "{{ docker_image_force_build }}"
 
 - name: Checkout db repo
   git:
@@ -96,6 +97,7 @@
     name: '{{ mantisbt_db_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-mysql-data'
+    force: "{{ docker_image_force_build }}"
 
 # *****************************************************************************
 # copy the MySQL config to the restore directory
diff --git a/roles/docker-mediawiki/tasks/main.yml b/roles/docker-mediawiki/tasks/main.yml
index a7be0a25..27e4a26a 100644
--- a/roles/docker-mediawiki/tasks/main.yml
+++ b/roles/docker-mediawiki/tasks/main.yml
@@ -82,6 +82,7 @@
     name: '{{ wiki_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-mediawiki'
+    force: "{{ docker_image_force_build }}"
 
 # *****************************************************************************
 # Create the data volumes
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index 8262a2aa..9c102e67 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -72,6 +72,7 @@
     name: '{{ openldap_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-openldap'
+    force: "{{ docker_image_force_build }}"
 
 # *****************************************************************************
 # Create the data volumes
diff --git a/roles/docker-openssl/tasks/main.yml b/roles/docker-openssl/tasks/main.yml
index bf716ae4..ed460998 100644
--- a/roles/docker-openssl/tasks/main.yml
+++ b/roles/docker-openssl/tasks/main.yml
@@ -65,6 +65,7 @@
     name: '{{ openssl_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-openssl'
+    force: "{{ docker_image_force_build }}"
 
 # *****************************************************************************
 # Create the data volumes
diff --git a/roles/docker-svn/tasks/main.yml b/roles/docker-svn/tasks/main.yml
index 8473ab13..575ce46c 100644
--- a/roles/docker-svn/tasks/main.yml
+++ b/roles/docker-svn/tasks/main.yml
@@ -65,6 +65,7 @@
     name: '{{ svn_image_name }}'
     tag: '{{ docker_image_tag }}'
     path: '{{ docker_projects_dir }}/docker-svn'
+    force: "{{ docker_image_force_build }}"
 
 # *****************************************************************************
 # Create the data volumes
